packages/components/src/navigation/index.js

```diff
@@ -57,13 +60,28 @@ export default function Navigation( {
 		}
 	}, [ activeItem, activeMenu ] );
 
+	useEffect( () => {
+		const items = findNavigationItems( children );
+		const menus = findNavigationMenus( children );
+
+		setNavigationItems( items );
+		setNavigationMenus( menus );
+	}, [] );
+
 	const context = {
 		activeItem: item,
 		activeMenu: menu,
 		setActiveItem,
 		setActiveMenu,
+		items: navigationItems,
+		menus: navigationMenus,
 	};
 
+	// Make sure menus and items are available before rendering
+	if ( ! navigationMenus || ! navigationItems ) {
+		return null;
+	}
```

Comment:
This is unnecessary I think.
If `Navigation` is empty, it wouldn't render anything anyway. 😄 

(...except the wrapper, and I guess it might be easier to debug an empty wrapper, than a completely unrendered component)

Reply:
`navigationMenus` and `navigationItems` are empty on the first render, the goal was to avoid empty objects in the context

Reply:
Removed this and made sure they are an empty object setting a value.